src/VisualStudio/Core/Def/Utilities/ClipboardHelpers.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace Microsoft.VisualStudio.LanguageServices.Utilities
+{
+    internal static class ClipboardHelpers
+    {
+        [DllImport("ole32.dll", ExactSpelling = true, CharSet = CharSet.Unicode)]
+        private static extern int OleGetClipboard(out IDataObject dataObject);
+
+        public static string? GetText()
+        {
+            IDataObject? dataObject = null;
+            var result = OleGetClipboard(out dataObject);
+            if (result != 0) // S_OK
+            {
+                // Report NFW?
+            }
+
+            if (dataObject is null || !dataObject.GetDataPresent(typeof(string)))
```

Comment:
Here you will have check for the TEXT/UNICODE_TEXT formats:

https://devdiv.visualstudio.com/DevDiv/_git/CPS?path=/src/Microsoft.VisualStudio.ProjectSystem.VS.Implementation/Package/CopyPaste/ClipboardFormat.cs&version=GBmain&line=19&lineEnd=24&lineStartColumn=9&lineEndColumn=54&lineStyle=plain&_a=contents.